File: fba/src/main/java/PostCdFormatter.java
public class PostCdFormatter {

	public String getPostCd(String postCd) {
		postCd = postCd.replaceAll("[\\-‐ー－−－ｰ 　]", "");
		postCd = String.format("%7s", postCd);
		postCd = postCd.replaceAll(" ", "0");
		return postCd;
	}
}

File: fba_kotlin/src/main/kotlin/PostCdFormatter.kt
import java.util.regex.Pattern

class PostCdFormatter {

    fun getPostCd(postCd: String): String {
        var postCd = postCd
        postCd = postCd.replace("[\\-‐ー－−－ｰ 　]".toRegex(), "")
        postCd = String.format("%7s", postCd)
        postCd = postCd.replace(" ".toRegex(), "0")
        return postCd
    }
}